protos/dcs/mission/v0/mission.proto

```diff
@@ -303,6 +303,49 @@ message StreamEventsResponse {
     dcs.common.v0.Airbase place = 3;
   }
 
+  /**
+   * Fired when a player connected to the server.
+   */
+  message ConnectEvent {
+    // The player's IP and port.
+    string addr = 1;
+    // The name of the player.
+    string name = 2;
+    // The player's unique client identifier (used to ban a player).
+    string ucid = 3;
+    // The player's id in the current server session (used to for name/slot/... changes).
+    uint32 id = 4;
+  }
+
+  /**
+   * The reason a player disconnected for.
+   */
+  enum DisconnectReason {
+    DISCONNECT_REASON_THATS_OKAY = 0;
+    DISCONNECT_REASON_INVALID_ADDRESS = 1;
+    DISCONNECT_REASON_CONNECT_FAILED = 2;
+    DISCONNECT_REASON_WRONG_VERSION = 3;
+    DISCONNECT_REASON_PROTOCOL_ERROR = 4;
+    DISCONNECT_REASON_TIMEOUT = 5;
+    DISCONNECT_REASON_INVALID_PASSWORD = 100;
```

Comment:
Do you still have a proto linter installed locally? If so, is the linter unhappy with the non-continuous numbering here (going from `5` to `100`)?

Reply:
Non-contiguous numbers is fine but it will complain about the use of 0. It expects

`DISCONNECT_REASON_INVALID = 0` is what it will expect.

I recommend shifting everything up by 1 like we did elsewhere

Reply:
Hm, I kinda liked the default to be `0`, which is the user leaving the server without an "error". How'd you feel about just renaming it to `DISCONNECT_REASON_UNSPECIFIED` and keeping the meaning of "manual disconnect" (aka no error)?

Reply:
My worry is that using 0 will cause a client who happens to use Language X extra pain in the future for no real technical reason. `Unspecified` is not really the same as `No error` so I am not entirely sure we can get away with calling them the same thing.

To avoid this potential issue requires a `reason + 1` to line 88 of the hook lua and incrementing the values here so I don't think we can claim a solid technical reason for not doing it.

It will also improve consistency since we have changed all the other ENUMs in the codebase to use 0 as unspecified.

Reply:
What if we call it `DISCONNECT_REASON_UNKNOWN` or `DISCONNECT_REASON_UNSPECIFIED`? The latter is probably more correct with the DCS API because users can just - you know - leave the server ( insert drum roll -- I'll show myself out :D )

Reply:
`DISCONNECT_REASON_UNSPECIFIED` and `DISCONNECT_REASON_THATS_OKAY` (i.e. no error) are not synonymous to me, which is what my previous comment was trying to get at.

Reply:
Done in c869bea9d67b289ba3f85910705b2d5b9b93e1f6 👍